plugins/catalog-react/report-alpha.api.md

```diff
@@ -324,6 +321,54 @@ export interface EntityContentLayoutProps {
   }>;
 }
 
+// @alpha (undocumented)
+export type EntityPredicate =
+  | EntityPredicateExpression
+  | EntityPredicatePrimitive
+  | {
+      $all: EntityPredicate[];
+    }
+  | {
+      $any: EntityPredicate[];
+    }
+  | {
+      $not: EntityPredicate;
+    };
+
+// @alpha (undocumented)
+export type EntityPredicateExpression = {
+  [KPath in string]: EntityPredicateValue;
+} & {
+  [KPath in `$${string}`]: never;
+};
+
+// @alpha (undocumented)
+export type EntityPredicatePrimitive = string | number | boolean;
+
+// @alpha
+export function entityPredicateToFilterFunction<T extends JsonValue>(
+  entityPredicate: EntityPredicate,
+): (value: T) => boolean;
+
+// @alpha (undocumented)
+export type EntityPredicateValue =
+  | EntityPredicatePrimitive
+  | {
+      $exists: boolean;
+    }
+  | {
+      $in: EntityPredicatePrimitive[];
+    }
+  | {
+      $contains: EntityPredicateExpression;
+    };
+
+// @alpha
+export function evaluateEntityPredicate(
```

Comment:
is it ok to force people to do `entityPredicateToFilterFunction(p)(value)` instead?

Reply:
ah yep let's clean this up